apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -34,26 +34,50 @@ function wpcom_should_limit_global_styles( $blog_id = 0 ) {
 		}
 	}
 
-	// Do not limit Global Styles on sites created before we made it a paid feature (2022-12-15).
-	if ( $blog_id < 213403000 ) {
-		return false;
+	// Limit Global Styles on sites with the 'wpcom-limit-global-styles' sticker.
+	if ( wpcom_global_styles_has_blog_sticker( 'wpcom-limit-global-styles', $blog_id ) ) {
+		return true;
 	}
 
-	if ( wpcom_site_has_feature( WPCOM_Features::GLOBAL_STYLES, $blog_id ) ) {
+	// Do not limit Global Styles on demo sites.
+	if ( wpcom_global_styles_has_blog_sticker( 'theme-demo-site', $blog_id ) ) {
 		return false;
 	}
 
-	if ( function_exists( 'has_blog_sticker' ) && has_blog_sticker( 'theme-demo-site', $blog_id ) ) {
+	// Do not limit Global Styles on sites created before we made it a paid feature (2022-12-15),
+	// that have never used Global Styles.
+	if (
+		$blog_id < 213403000 &&
+		defined( 'IS_WPCOM' ) && IS_WPCOM && ! wpcom_global_styles_has_been_used( $blog_id )
```

Comment:
Shouldn't this condition be inverted? From what I understand, this it should **not** limit the GS if there is no usage of GS.

Reply:
Yes, yes it should!